networkx/algorithms/centrality/subgraph_alg.py

```diff
@@ -74,10 +74,11 @@ def subgraph_centrality_exp(G):
     ['1 3.90', '2 3.90', '3 3.64', '4 3.71', '5 3.64', '6 3.71', '7 3.64', '8 3.90']
     """
     # alternative implementation that calculates the matrix exponential
+    import numpy
     import scipy.linalg
-    nodelist = list(G) # ordering of nodes in matrix
-    A = nx.to_numpy_matrix(G,nodelist)
-    # convert to 0-1 matrix
+    nodelist = list(G) # ordering of nodes in array
+    A = nx.to_numpy_recarray(G,nodelist,dtype=numpy.dtype(int))
+    # convert to 0-1 array
     A[A!=0.0] = 1
     expA = scipy.linalg.expm(A)
```

Comment:
Sorry, I meant use the original code with none of your changes.  Then say `scipy.linalg.expm(A.A)` instead of  `scipy.linalg.expm(A)` .